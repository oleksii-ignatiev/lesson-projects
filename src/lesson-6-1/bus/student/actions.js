import { types } from './types';

export const studentActions = {
    fillProfile: (profile) => {
        return {
            type: types.USER_FILL,
            payload: profile
        }
    }
}
