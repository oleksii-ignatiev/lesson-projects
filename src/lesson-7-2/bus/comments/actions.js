import { types } from './types';

import { api } from '../../api';

export const commentsActions = Object.freeze({

    startFetching: () => {
        return {
            type: types.COMMENTS_START_FETCHING
        }
    },
    stopFetching: () => {
        return {
            type: types.COMMENTS_STOP_FETCHING
        }
    },
    fill: (payload) => {
        return {
            type: types.COMMENTS_FILL,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.COMMENTS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },

    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.COMMENTS_FETCH_ASYNC
        });

        dispatch(commentsActions.startFetching());

        const response = await api.comments.fetch();
    
        if (response.status === 200) {
            const results = await response.json();

            dispatch(commentsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(commentsActions.setFetchingError(error));
        }
        dispatch(commentsActions.stopFetching());
    }
});