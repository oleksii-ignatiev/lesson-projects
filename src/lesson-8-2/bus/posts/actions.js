import { types } from './types';

import { api } from '../../api';

export const postsActions = Object.freeze({

    startFetching: () => {
        return {
            type: types.POSTS_START_FETCHING
        }
    },
    stopFetching: () => {
        return {
            type: types.POSTS_STOP_FETCHING
        }
    },
    fill: (payload) => {
        return {
            type: types.POSTS_FILL,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.POSTS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },

    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.POSTS_FETCH_ASYNC
        });

        dispatch(postsActions.startFetching());

        const response = await api.posts.fetch()
    
        if (response.status === 200) {
            const results = await response.json();

            dispatch(postsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(postsActions.setFetchingError(error));
        }
        dispatch(postsActions.stopFetching());
    }
});