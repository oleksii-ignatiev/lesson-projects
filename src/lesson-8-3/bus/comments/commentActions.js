import { types } from './types';

import { api } from '../../api';

export const commentActions = Object.freeze({

    startFetching: () => {
        return {
            type: types.COMMENT_START_FETCHING
        }
    },
    stopFetching: () => {
        return {
            type: types.COMMENT_STOP_FETCHING
        }
    },
    fill: (payload) => {
        return {
            type: types.COMMENT_FILL,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.COMMENT_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },

    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.COMMENT_FETCH_ASYNC
        });

        dispatch(commentActions.startFetching());

        const response = await api.comment.fetch(id);
        
        if (response.status === 200) {
            const results = await response.json();

            dispatch(commentActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(commentActions.setFetchingError(error));
        }
        dispatch(commentActions.stopFetching());
    }
});