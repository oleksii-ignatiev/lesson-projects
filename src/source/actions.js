import { types } from './types';

import { api } from './api/api';

export const forcastActions = Object.freeze({

    startFetching: () => {
        return {
            type: types.FORCAST_START_FETCHING
        }
    },
    stopFetching: () => {
        return {
            type: types.FORCAST_STOP_FETCHING
        }
    },
    fill: (payload) => {
        return {
            type: types.FORCAST_FILL,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FORCAST_SET_FETCHING_ERROR,
            isError: true,
            payload: error
        }
    },

    fetchAsync: (days) => async (dispatch) => {
        dispatch({
            type: types.FORCAST_FETCH_ASYNC
        });

        dispatch(forcastActions.startFetching());

        const response = await api.forcast.get(days);
        
        if (response.status === 200) {
            const results = await response.json();
            
            dispatch(forcastActions.fill(results.data));
            
        } else {
            const error = {
                status: response.status
            };

            dispatch(forcastActions.setFetchingError(error));
        }
        dispatch(forcastActions.stopFetching());
    },

    setIsSubmit: (payload) => {
        return {
            type: types.IS_SUBMIT_SET,
            payload: payload
        }
    },

    setIsChecked: (payload) => {
        return {
            type: types.IS_CHECKED_SET,
            payload: payload
        }
    },

    setMaxTemp: (payload) => {
        return {
            type: types.MAX_TEMPERATURE_SET,
            payload: payload
        }
    },

    setMinTemp: (payload) => {
        return {
            type: types.MIN_TEMPERATURE_SET,
            payload: payload
        }
    },

    setButtonValue: (payload) => {
        return {
            type: types.BUTTON_VALUE_SET,
            payload: payload
        }
    },

    setInitState: () => {
        return {
            type: types.FORCAST_INIT_STATE
        }
    }
});