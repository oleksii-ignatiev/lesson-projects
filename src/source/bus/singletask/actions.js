import { types } from './types';

import { api } from '../../api';

export const formActions = Object.freeze({

    startEditing: () => {
        return {
            type: types.FORM_START_EDITING
        }
    },
    // startFetching: () => {
    //     return {
    //         type: types.FORM_START_FETCHING
    //     }
    // },
    // stopFetching: () => {
    //     return {
    //         type: types.FORM_STOP_FETCHING
    //     }
    // },
    fill: (payload) => {
        return {
            type: types.FORM_POST_DATA,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FORM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },

    fetchAsync: (type) => async (dispatch) => {
        dispatch({
            type: type
        });

        dispatch(formActions.startFetching());
        let response;
        switch (type) {
            case 'FORM_POST_DATA': response = await api.post.create();
            case 'FORM_GET_DATA': response = await api.post.get();
            case 'FORM_UPDATE_DATA': response = await api.post.update();
            case 'FORM_POST_DATA': response = await api.post.create();
            case 'FORM_POST_DATA': response = await api.post.create();
            case 'FORM_POST_DATA': response = await api.post.create();
            case 'FORM_POST_DATA': response = await api.post.create();

        }
            
        if (response.status === 200) {
            const results = await response.json();

            dispatch(formActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(formActions.setFetchingError(error));
        }
        // dispatch(formActions.stopFetching());
    }
});
