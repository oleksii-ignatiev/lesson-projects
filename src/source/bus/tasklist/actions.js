import { types } from './types';

import { api } from '../../api';

export const tasksActions = Object.freeze({

    startFetching: () => {
        return {
            type: types.TASKS_START_FETCHING
        }
    },
    stopFetching: () => {
        return {
            type: types.TASKS_STOP_FETCHING
        }
    },
    fill: (payload) => {
        return {
            type: types.TASKS_FILL,
            payload: payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.TASKS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },

    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.TASKS_FETCH_ASYNC
        });

        dispatch(tasksActions.startFetching());

        const response = await api.task.get();
        
            
        if (response.status === 200) {
            const results = await response.json();
            
            dispatch(tasksActions.fill(results.data));
            
        } else {
            const error = {
                status: response.status
            };

            dispatch(tasksActions.setFetchingError(error));
        }
        dispatch(tasksActions.stopFetching());
    },

// Single task actions
startTaskFetching: () => {
    return {
        type: types.TASK_START_FETCHING
    }
},
stopTaskFetching: () => {
    return {
        type: types.TASK_STOP_FETCHING
    }
},
fillTask: (payload) => {
    return {
        type: types.TASK_FILL,
        payload: payload
    }
},
setTaskFetchingError: (error) => {
    return {
        type: types.TASK_SET_FETCHING_ERROR,
        error: true,
        payload: error
    }
},

fetchTaskAsync: (hash) => async (dispatch) => {
    dispatch({
        type: types.TASK_FETCH_ASYNC
    });

    dispatch(tasksActions.startTaskFetching());

    const response = await api.task.getTask(hash);
    
        
    if (response.status === 200) {
        const results = await response.json();
        
        dispatch(tasksActions.fillTask(results.data));
    } else {
        const error = {
            status: response.status
        };

        dispatch(tasksActions.setTaskFetchingError(error));
    }
    dispatch(tasksActions.stopTaskFetching());
}

});
