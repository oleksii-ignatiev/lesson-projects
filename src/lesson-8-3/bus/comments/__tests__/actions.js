import { commentsActions } from "../actions";
import { commentActions } from "../commentActions";

describe('Comments Actions', () => {

    // comments actions
    test('Comments startFetching', () => {
        expect(commentsActions.startFetching()).toMatchSnapshot();
    });
    test('Comments stopFetching', () => {
        expect(commentsActions.stopFetching()).toMatchSnapshot();
    });
    test('Comments fetchAsync', () => {
        expect(commentsActions.fetchAsync()).toMatchSnapshot();
    });
    test('Comments fill', () => {
        expect(commentsActions.fill({name: 'Comments'})).toMatchSnapshot();
    });
    test('Comments setFetchingError', () => {
        expect(commentsActions.setFetchingError('We have a problem')).toMatchSnapshot();
    });

    // comment actions
    test('Comment startFetching', () => {
        expect(commentActions.startFetching()).toMatchSnapshot();
    });
    test('Comment stopFetching', () => {
        expect(commentsActions.stopFetching()).toMatchSnapshot();
    });
    test('Comment fetchAsync', () => {
        expect(commentsActions.fetchAsync()).toMatchSnapshot();
    });
    test('Comment fill', () => {
        expect(commentsActions.fill({name: 'Comment'})).toMatchSnapshot();
    });
    test('Comment setFetchingError', () => {
        expect(commentsActions.setFetchingError('We have a problem')).toMatchSnapshot();
    });
});
