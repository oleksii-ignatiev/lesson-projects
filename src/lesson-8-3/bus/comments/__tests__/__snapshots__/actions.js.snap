// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Actions Comment fetchAsync 1`] = `[Function]`;

exports[`Comments Actions Comment fill 1`] = `
Object {
  "payload": Object {
    "name": "Comment",
  },
  "type": "COMMENTS_FILL",
}
`;

exports[`Comments Actions Comment setFetchingError 1`] = `
Object {
  "error": true,
  "payload": "We have a problem",
  "type": "COMMENTS_SET_FETCHING_ERROR",
}
`;

exports[`Comments Actions Comment startFetching 1`] = `
Object {
  "type": "COMMENT_START_FETCHING",
}
`;

exports[`Comments Actions Comment stopFetching 1`] = `
Object {
  "type": "COMMENTS_STOP_FETCHING",
}
`;

exports[`Comments Actions Comments fetchAsync 1`] = `[Function]`;

exports[`Comments Actions Comments fill 1`] = `
Object {
  "payload": Object {
    "name": "Comments",
  },
  "type": "COMMENTS_FILL",
}
`;

exports[`Comments Actions Comments setFetchingError 1`] = `
Object {
  "error": true,
  "payload": "We have a problem",
  "type": "COMMENTS_SET_FETCHING_ERROR",
}
`;

exports[`Comments Actions Comments startFetching 1`] = `
Object {
  "type": "COMMENTS_START_FETCHING",
}
`;

exports[`Comments Actions Comments stopFetching 1`] = `
Object {
  "type": "COMMENTS_STOP_FETCHING",
}
`;
